src/mercury_prof_interface.c

```diff
@@ -0,0 +1,281 @@
+ * found at the root of the source code distribution tree.
+ */
+
+#include "mercury.h"
+#include "mercury_bulk.h"
+#include "mercury_core.h"
+#include "mercury_private.h"
+#include "mercury_error.h"
+
+#include "mercury_atomic.h"
+#include "mercury_types.h"
+#include "mercury_prof_interface.h"
+#include "mercury_prof_pvar_impl.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+/************************************/
+/* Local Type and Struct Definition */
+/************************************/
+
+/* HG profiling PVAR handle object concrete definition */
+struct hg_prof_pvar_handle {
+   hg_prof_class_t pvar_class; /* PVAR class */
+   hg_prof_datatype_t pvar_datatype; /* PVAR datatype */
+   hg_prof_bind_t pvar_bind; /* PVAR binding */
+   int continuous; /* is PVAR continuous or not */
+   void * addr; /* actual address of PVAR */
+   int is_started; /* if continuous, has PVAR been started? */
+   int count; /* number of values associated with PVAR */
+   char name[128]; /* PVAR name */
+   char description[128]; /* PVAR description */
+};
+
+/* HG profiling session object concrete definition */
+struct hg_prof_pvar_session {
+   hg_prof_pvar_handle_t * pvar_handle_array; /* PVAR handle array */
+   int is_initialized;     /* Is profiling initialized */
+};
+
+/* HG class */
+struct hg_private_class {
+    struct hg_class hg_class;       /* Must remain as first field */
+    int hg_prof_is_initialized;     /* Is profiling initialized */
+    int num_pvars;          /* No of PVARs currently supported */
+    hg_prof_pvar_session_t session; /* Is profiling initialized on the session */
+};
+
+/*******************/
+/* Local Variables */
+/*******************/
+
+/*---------------------------------------------------------------------------*/
+static void 
+hg_prof_set_is_initialized(struct hg_private_class * hg_private_class)
+{
+  hg_private_class->hg_prof_is_initialized = 1;
+}
+
+/*---------------------------------------------------------------------------*/
+static int 
+hg_prof_get_is_initialized(struct hg_private_class * hg_private_class) {
+   return hg_private_class->hg_prof_is_initialized;
+}
+
+/*---------------------------------------------------------------------------*/
+static void 
+hg_prof_set_session_is_initialized(struct hg_prof_pvar_session * session)
+{
+  session->is_initialized = 1;
+}
+
+/*---------------------------------------------------------------------------*/
+static int 
+hg_prof_get_session_is_initialized(struct hg_prof_pvar_session * session) {
+   return session->is_initialized;
+}
+
+/*---------------------------------------------------------------------------*/
+hg_return_t 
+HG_Prof_init(hg_class_t *hg_class) {
+
+  struct hg_private_class *hg_private_class = (struct hg_private_class *)(hg_class);
+
+  hg_prof_set_is_initialized(hg_private_class);
+  hg_private_class->num_pvars = NUM_PVARS;
+  hg_private_class->session = NULL;
+
```

Comment:
I don't think you are allowed to do that here :) HG_Prof_init() should return a new hg_prof_class_t * so I would expect you to malloc a new struct hg_prof_class in that routine, fill it and return it.